### PR TITLE
Config fixes and updates

### PR DESCRIPTION
**Pull request by supertick**
* fixes the ability to save and preserve servo values in InMoov2
* fixes null locales
* fixes many npe's
* handles virtual attribute appropriately
* removes name from config - it's now "normalized" with the file name
* cleans out invalid notify entries
* doesnt npe on null locale when creating a chatbot
* removes machine generated attaches from config
* has explicit attaching 
* new AbstractSpeechRecognizer config

### CODE REVIEW

File: src/main/java/org/myrobotlab/service/config/ServiceConfig.java [hunk 1]
```diff
@@ -1,19 +1,17 @@
 package org.myrobotlab.service.config;
 
-import java.util.ArrayList;
-
 public class ServiceConfig {
 
-  public String name;
+  // public String name; not normalized
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/config/ServiceConfig.java]: Why are we removing name ?  all services have a name and a type...

File: src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechRecognizer.java [hunk 2]
```diff
@@ -572,4 +560,29 @@ public void unsetWakeWord() {
     setWakeWord(null);
   }
 
+  @Override
+  public ServiceConfig getConfig() {
+    WebkitSpeechRecognitionConfig config = (WebkitSpeechRecognitionConfig) initConfig(new WebkitSpeechRecognitionConfig());
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechRecognizer.java]: Should be AbstractSpeechRecognizerConfig not webkit..

File: src/main/java/org/myrobotlab/service/config/ServiceConfig.java (hunk 1 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/config/ServiceConfig.java]: The name is and will always be in runtime.yml
runtime has a list of names of services to create
that name is tied to a filename to configure it
the name inside the file is not normalized and deceiving

File: src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechRecognizer.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechRecognizer.java]: oop yep - fixed

**Pull request merged**